jetbrains-core/src/software/aws/toolkits/jetbrains/services/codewhisperer/language/CodeWhispererLanguageManager.kt

```diff
@@ -4,43 +4,35 @@
 package software.aws.toolkits.jetbrains.services.codewhisperer.language
 
 import com.intellij.openapi.components.service
-import software.aws.toolkits.jetbrains.services.codewhisperer.model.ProgrammingLanguage
-import software.aws.toolkits.telemetry.CodewhispererLanguage
+import com.intellij.openapi.vfs.VirtualFile
+import com.intellij.psi.PsiFile
+import software.aws.toolkits.jetbrains.services.codewhisperer.language.languages.CodeWhispererJava
+import software.aws.toolkits.jetbrains.services.codewhisperer.language.languages.CodeWhispererJavaScript
+import software.aws.toolkits.jetbrains.services.codewhisperer.language.languages.CodeWhispererJsx
+import software.aws.toolkits.jetbrains.services.codewhisperer.language.languages.CodeWhispererPlainText
+import software.aws.toolkits.jetbrains.services.codewhisperer.language.languages.CodeWhispererPython
+import software.aws.toolkits.jetbrains.services.codewhisperer.language.languages.CodeWhispererUnknownLanguage
 
 class CodeWhispererLanguageManager {
-    private val supportedLanguage = setOf(
-        CodewhispererLanguage.Java.toString(),
-        CodewhispererLanguage.Python.toString(),
-        CodewhispererLanguage.Javascript.toString()
-    )
-
-    fun isLanguageSupported(language: ProgrammingLanguage): Boolean {
-        val mappedLanguage = getParentLanguage(language)
-        return supportedLanguage.contains(mappedLanguage.languageName)
+    fun getLanguage(vFile: VirtualFile): CodeWhispererProgrammingLanguage {
+        val fileTypeName = vFile.fileType.name.lowercase()
+        return when {
+            fileTypeName.contains("python") -> CodeWhispererPython.INSTANCE
+            fileTypeName.contains("javascript") -> CodeWhispererJavaScript.INSTANCE
+            fileTypeName.contains("java") -> CodeWhispererJava.INSTANCE
+            fileTypeName.contains("jsx") -> CodeWhispererJsx.INSTANCE
```

Comment:
does this need to be "jsx harmony"?

Reply:
yep it should be updated for the tsx file

Reply:
current version will support `.tsx`

Reply:
yeah I was thinking if we update here we don't need #3318 lol

Reply:
yeah, with this PR, it will a lot easier to add new languages and maintain lol

Reply:
it's error prone to convert those string back and forward

Reply:
<img width="1393" alt="tsx" src="https://user-images.githubusercontent.com/96078566/192911889-4f8d3655-e9ff-472b-b537-6769bb5c3ff4.png">

Reply:
alright, fixed it

Reply:
thanks for the refactoring. Much better than the back and forth